cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1676,7 +1679,34 @@ func generateMountsModeRun(mst *initramfsMountsState) (string, error) {
 		}
 	}
 
-	// 4.4 mount snapd snap only on first boot
+	// 4.4 check if we expected a ubuntu-seed partition from the gadget data
+	if isClassic {
+		gadgetDir := filepath.Join(boot.InitramfsRunMntDir,
+			snapTypeToMountDir[snap.TypeGadget])
+		gadgetInfo, err := gadget.ReadInfo(gadgetDir, model)
+		if err != nil {
+			return "", err
+		}
+		seedDefinedInGadget := false
+	volLoop:
+		for _, vol := range gadgetInfo.Volumes {
+			for _, part := range vol.Structure {
+				if part.Role == gadget.SystemSeed &&
```

Comment:
checking role is enough I think, we check the part Name when we validate the gadget I think